src/EditorFeatures/Core/Implementation/Suggestions/SuggestedAction.cs

```diff
@@ -138,22 +178,24 @@ public virtual object GetPreview(CancellationToken cancellationToken)
                     var preferredDocumentId = Workspace.GetDocumentIdInCurrentContext(SubjectBuffer.AsTextContainer());
                     var preferredProjectid = preferredDocumentId == null ? null : preferredDocumentId.ProjectId;
 
-                    return previewResult.TakeNextPreview(preferredDocumentId, preferredProjectid);
+                    // TakeNextPreviewAsync() needs to run on UI thread.
+                    AssertIsForeground();
+                    return await previewResult.TakeNextPreviewAsync(preferredDocumentId, preferredProjectid, cancellationToken).ConfigureAwait(true);
                 }
-            });
 
-            var optionService = Workspace.Services.GetService<IOptionService>();
-            if (optionService == null || !optionService.GetOption(InternalFeatureOnOffOptions.EnhancedPreviewPane))
-            {
-                return previewContent;
-            }
+                // GetPreviewPane() below needs to run on UI thread. We use ConfigureAwait(true) to stay on the UI thread.
+            }).ConfigureAwait(true);
 
             var previewPaneService = Workspace.Services.GetService<IPreviewPaneService>();
             if (previewPaneService == null)
             {
-                return previewContent;
+                return null;
```

Comment:
No. This is an async method which means we can return T and compiler will do its magic to wrap a Task<T> around it.